Dockerfile

```diff
@@ -0,0 +1,20 @@
+#
+# Creates a configuration of the open source Docker registry
+# suitable for using with Synapse
+#
+#
+FROM registry:2
+# stack is dev or prod
+ARG stack=dev
+
+# switches between 'dev' and 'prod'
+COPY resources/${stack}/config.yml /etc/docker/registry/config.yml
+COPY resources/${stack}/token_signing_key_public_cert.pem /etc/docker/registry/token_signing_key_public_cert.pem
+# the self-signed cert' and private key should be generated before running 'docker build'
+# see the github workflow for an example of how to do this
+COPY privatekey.pem /etc/docker/registry/ssl/privatekey.pem
+COPY certificate.pem /etc/docker/registry/ssl/certificate.pem
```

Comment:
I don't really understand how these will provide secrets for the docker registry.

Reply:
The Docker registry needs these files and this step in the container build copies the files from the host (e.g., the machine running the GitHub workflow) into the container image.  Is that what you are asking about?

Reply:
i get it now, your answer to my question `i don't see how the secret is inserted into the registry configuration file. could you please explain?` cleared it up.